### ISSUE TITLE
tour step placement bottom-right don t work

### ISSUE THREAD

**Issue by brusand** [benmarch/angular-ui-tour#136]
**I am using the _exact_ versions of the following:**
Browser: [Chome|Firefox|IE|Safari] Version 58.0.3029.110 (64-bit)
AngularJS: "angular": "1.5.5"
Angular Bootstrap:" angular-ui-bootstrap": "1.3.3"
Angular UI Tour: "angular-ui-tour": "^0.8.1",
I have installed this library via: (NPM, Bower, or downloaded package)
npm
I have observed the following behavior:

when i set popover placement to bottom-right, the arrow is still in the center position but the box is well bottom right align

i observe some flipping 

(all go well with bottom and bottom-left placements)

This is how I expected it to behave:
i want the arrow display on the top right corner of the popover
 
**Here is my tour config, and all related step configs:**

```js
```
```html
    <div
      ui-tour="welcome"
      ui-tour-popup-class="onboarding"
      ui-tour-template-url="'onboarding.tpl.html'"
      ui-tour-backdrop="true"
      class="collapse navbar-collapse" id="main-menu">
```
```html
          <div
            tour-step="account"
            tour-step-title="{{'lo.onboarding.configuration.title' | translate }}"
            tour-step-content="{{'lo.onboarding.configuration.description' | translate }}"
            tour-step-placement="bottom-right">
          </div>
```
**Additional notes/code:**

i can send you in private a screen capture if you want ?
thx
Bruno

**Comment by benmarch** [comment]
Hey Bruno,

Would you mind attaching a screenshot or two? I have included some CSS in the library to accommodate for [arrow positioning](https://github.com/benmarch/angular-ui-tour/blob/master/app/styles/angular-ui-tour.css#L27), so I'm not really sure why it isn't working. A screenshot with the issue, as well as the CSS for the popover and the arrow would be really helpful.

Thanks,
Ben

**Comment by brusand** [comment]
Email me bruno.darrigues@gmail.com, i ll send you  à screen captur.   :)

**Comment by benmarch** [comment]
Hey Bruno, I found the issue: Bootstrap is setting the `left` property of the arrow, and I'm setting the `right`. I will patch it up and release a new version. In the meantime, you can fix it by setting the following CSS:

```css
.popover.bottom > .arrow {
  left: auto;
}
```

You might need to add specificity or `!important` if it doesn't take.

**Comment by brusand** [comment]
many thx Ben

it works much more better  :)

Bruno

2017-06-13 16:43 GMT+02:00 Ben March <notifications@github.com>:

**Comment by brusand** [comment]
hi Ben,

Sorry but i have still an issue with bottom-right.
i add your fixed

.popover.bottom> .arrow {
  left: auto;
}

my html is like this : tour-step account with bottom-right placement is ok

    <li cmn-state-active="lo.app.auth.config"
ng-show="!$ctrl.rights.SUPER_ADMIN">
      <a id="configuration-nav-link" href="#/account">
        <span translate="lo.configuration.title">Configuration</span>
      </a>
      <div
        tour-step="account"
        tour-step-title="{{'lo.onboarding.configuration.title' | translate }}"
        tour-step-content="{{'lo.onboarding.configuration.description'
| translate }}"
        tour-step-placement="bottom-right">
      </div>
    </li>
    <li cmn-state-active="lo.app.auth.tools">
      <a id="simulation-nav-link" href="#/tools"
ng-show="$ctrl.rights.DEV.READ && !$ctrl.rights.TENANT.WRITE">
        <span translate="lo.dev_tools.title">Simulation</span>
      </a>
    </li>
    

  </ul>
  <lo-account-nav></lo-account-nav>
</div>


next tour step in lo-account-nav component FAILED

<ul class="nav navbar-nav navbar-right">
  <li cmn-state-active="lo.app.sdk">
    <a href="#/sdk" translate="lo.sdk.title">IoT Soft Box</a>
    <div ng-if="$ctrl.loggedIn"
         tour-step="sdk"
         tour-step-title="{{'lo.onboarding.iotsoftbox.title' | translate }}"
         tour-step-content="{{'lo.onboarding.iotsoftbox.description' |
translate }}"
         tour-step-placement="bottom"
    />
  </li>

my last item menu (account /  Configuration) is set with bottom-right
placement and it is OK :
[image: Images intégrées 1]

the next tour-step is in an sub component and set to bottom-right placement
(Iot Soft Box) and it is FAILED

[image: Images intégrées 6]

if the above failed sub item, is set to bottom, it is ko too : placement is
ok but not the arrow (not in center) :
[image: Images intégrées 7]

2017-06-13 16:43 GMT+02:00 Ben March <notifications@github.com>:

**Comment by benmarch** [comment]
This is fixed in 0.8.2.
